python/ray/serve/pipeline/ingress.py

```diff
@@ -0,0 +1,32 @@
+import starlette
+from importlib import import_module
+
+from ray import serve
+from ray.experimental.dag import DAGNode
+from ray.serve.utils import parse_import_path, _get_logger
+
+logger = _get_logger()
+
+
+@serve.deployment
```

Comment:
should probably just be a class?

Reply:
that works too, only change needed is to make ingress generation like `serve.deployment(Ingress)` instead. I think this is more favorable given we register serve decorated classes, and better off not providing any in our codebase.